### PR TITLE
Refactor trace fixture mutation with serde_json

### LINKED ISSUE #28: refactor(tests): replace brittle string patching with structured JSON mutation in worker_coverage.rs
## Background

`tests/e2e_traces/worker_coverage.rs` uses string-level patching to mutate fixture data. This was flagged as a deferred refactor during PR #23 (https://github.com/leynos/axinite/pull/23).

## Proposed Change

Replace the brittle string-replacement approach with structured JSON mutation using `serde_json` — for example, deserialise the fixture into a `serde_json::Value`, apply the required field mutations, and re-serialise. This can be wrapped in a small helper function to keep test bodies concise.

## Motivation

String patching is fragile: it silently fails when the fixture format changes and is hard to reason about. Structured mutation is type-safe, self-documenting, and resilient to whitespace or key-ordering changes in fixtures.

## References

- PR #23: https://github.com/leynos/axinite/pull/23

### PR DESCRIPTION
**Pull request by leynos**
## What changed
- added a reusable `load_trace_with_mutation` helper that loads a trace fixture into `serde_json::Value`, applies a caller-supplied mutation, and deserializes back to `LlmTrace`
- updated `worker_coverage::tool_error_feedback` to mutate the recovery-path field through an explicit JSON path instead of `patch_path` string replacement
- hardened `advanced_traces::tool_error_recovery` by waiting for a successful `write_file` completion record before reading the recovery file

## Why it changed
String-level fixture patching is brittle and can silently stop matching when fixture formatting or surrounding content changes. Structured JSON mutation makes the intended fixture change explicit and fails loudly if the fixture shape drifts.

The aggregate gate also exposed a timing race in the existing recovery test, so that test now waits for the successful tool completion before asserting on-disk state.

## Impact
- trace-based tests are less fragile around fixture formatting and key ordering
- the worker coverage case documents the exact fixture field it depends on
- the advanced recovery trace is stable under the full test gate

## Validation
- `make all`
- `cargo test --features test-helpers --test e2e_traces advanced_traces::tool_error_recovery -- --exact`
- `cargo test --features test-helpers --test e2e_traces worker_coverage::tool_error_feedback -- --exact`

## Summary by Sourcery

Refactor trace-based tests to use structured JSON mutation and stabilize tool error recovery timing.

New Features:
- Introduce a reusable helper to load trace fixtures as JSON, apply structured mutations, and deserialize back to LlmTrace.

Enhancements:
- Update worker coverage trace tests to mutate the recovery path via an explicit JSON path instead of string replacement.
- Ensure advanced trace recovery tests wait for a successful write_file completion before asserting file contents.
- Expose the new trace mutation helper through the trace LLM support module and validate its signature in test support.

Closes #28

## Summary by Sourcery

Refactor trace-based tests to use structured JSON mutation for fixtures and stabilize tool error recovery timing.

Enhancements:
- Expose a reusable helper to load trace fixtures as JSON, apply a mutation, and deserialize back to LlmTrace through the trace LLM support module.

Tests:
- Update the worker coverage tool_error_feedback test to mutate the recovery path via a structured JSON access instead of string replacement.
- Make the advanced tool_error_recovery trace test wait for a successful write_file completion before validating on-disk recovery output.
- Extend trace support symbol checks to cover the new trace mutation helper signature.

### CODE REVIEW

File: tests/e2e_traces/advanced_traces.rs [hunk 1]
```diff
@@ -119,14 +119,29 @@ async fn tool_error_recovery() {
         "expected 2 write_file calls (bad + good), got {write_count}"
     );
 
+    // Wait for the successful recovery write to be recorded before reading.
+    let completed = tokio::time::timeout(Duration::from_secs(5), async {
+        loop {
+            let completed = rig.tool_calls_completed();
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/e2e_traces/advanced_traces.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[tests/support/test_channel.rs](https://github.com/leynos/axinite/pull/161/files/a63524b90c8306a307d1ac1b39da866881cad962#diff-718bf2f29d24070e23408f799a82c359771b5ada23adbef2090cbbe3e7d47141)

Comment on lines +180 to +189

~~~rust
    pub async fn tool_calls_completed_async(&self) -> Vec<(String, bool)> {
        self.captured_status_events_async()
            .await
            .iter()
            .filter_map(|s| match s {
                StatusUpdate::ToolCompleted { name, success, .. } => Some((name.clone(), *success)),
                _ => None,
            })
            .collect()
    }
~~~

❌ New issue: [Code Duplication](https://codescene.io/projects/77987/delta?repo-id=1501128&review-id=161&biomarker=Code+Duplication&filename=tests%2Fsupport%2Ftest_channel.rs&method=)
The module contains 4 functions with similar structure: TestChannel.tool_calls_completed,TestChannel.tool_calls_completed_async,TestChannel.tool_calls_started,TestChannel.tool_results

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix. Include the file and symbol names indicated in the issue at the head of your response.

[tests/support/mod.rs](https://github.com/leynos/axinite/pull/161/files/a63524b90c8306a307d1ac1b39da866881cad962#diff-55805875b5522b6f9fb284322ec0d3768f380a00d50f3089999942ed9a73ba65)

Comment on lines +329 to +336

~~~rust
    touch_const!(
        for<'a> fn(&'a test_rig::TestRig) -> AsyncCompletedToolCalls<'a> =
            _tool_calls_completed_async_sig
    );
    touch_const!(
        for<'a> fn(&'a test_rig::TestRig) -> AsyncStatusEvents<'a> =
            _captured_status_events_async_sig
    );
~~~

❌ New issue: [Complex Method](https://codescene.io/projects/77987/delta?repo-id=1501128&review-id=161&biomarker=Complex+Method&filename=tests%2Fsupport%2Fmod.rs&method=touch_test_rig_async_sigs)
touch_test_rig_async_sigs has a cyclomatic complexity of 10, threshold = 9

**Comment by leynos** [comment]
@coderabbitai pause

File: tests/support/mod.rs [hunk 2]
```diff
@@ -152,6 +157,7 @@ fn trace_support_symbol_refs() {
     assert_trace_llm_from_file_async(trace_llm::TraceLlm::from_file_async);
     let _: fn(String) -> _ = trace_llm::LlmTrace::from_file_async;
     assert_trace_from_file_async(trace_llm::LlmTrace::from_file_async);
+    let _: fn(String, fn(&mut serde_json::Value)) -> _ = trace_llm::load_trace_with_mutation;
```
[the remark this answers is not in the document]
Comment (leynos) [reply tests/support/mod.rs:160]: @coderabbitai the engineer observes:

> The assertion is still present in tests/support/mod.rs, and `load_trace_with_mutation` is still defined as an `async fn` in tests/support/trace\_types.rs. That review finding is stale: an `async fn` can still be referenced as a function item and coerced to a function pointer with an inferred return type, which is exactly what `fn(String, fn(&mut serde_json::Value)) -> _` is doing here. The same pattern is already used just above for the other async loaders.
> 
> I rechecked this by compiling the affected target with `cargo check --features test-helpers --test e2e_traces`, and it passed cleanly. Since the code compiles as written and the assertion remains valid, I did not change anything.

**Comment by leynos** [comment]
@coderabbitai Have the following failed checks now been resolved?

If further work is required, please provide an AI agent prompt for the remaining work to be done to address these failures.

### ❌ Failed checks (3 warnings)

| Check name | Status | Explanation | Resolution |
| :-: | :-- | :-- | :-- |
| Testing | ⚠️ Warning | New public functions lack dedicated unit tests; only e2e integration tests exercise them, missing isolated coverage for error paths and edge cases. | Establish #\[cfg(test)\] modules for each public function testing success paths, error conditions, empty inputs, and concurrent access before merging. |
| Developer Documentation | ⚠️ Warning | Pull request introduces three new test APIs without corresponding documentation in developer guides. | Document load\_trace\_with\_mutation, tool\_calls\_completed\_async, and captured\_status\_events\_async in docs/developers-guide.md with usage examples and implementation references. |
| Module-Level Documentation | ⚠️ Warning | The file tests/support/test\_rig/rig.rs lacks module-level documentation; six of seven modified files include proper //! docstrings whilst this file omits the required module-level explanation. | Add a module-level //! docstring to tests/support/test\_rig/rig.rs before all imports, explaining the module's purpose and TestRig struct functionality, then verify with cargo doc --test --no-deps. |
